plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2711,6 +2748,77 @@ public void finishStatisticsCollection(ConnectorSession session, ConnectorTableH
                 Optional.of(physicalColumnNameMapping));
     }
 
+    private void generateMissingFileStatistics(ConnectorSession session, DeltaLakeTableHandle tableHandle, Collection<ComputedStatistics> computedStatistics)
+    {
+        Map<String, AddFileEntry> addFileEntriesWithNoStats = transactionLogAccess.getActiveFiles(
+                        getSnapshot(tableHandle.getSchemaTableName(), tableHandle.getLocation(), session), session)
+                .stream()
+                .filter(addFileEntry -> addFileEntry.getStats().isEmpty()
+                        || addFileEntry.getStats().get().getNumRecords().isEmpty()
+                        || addFileEntry.getStats().get().getMaxValues().isEmpty()
+                        || addFileEntry.getStats().get().getMinValues().isEmpty()
+                        || addFileEntry.getStats().get().getNullCount().isEmpty())
+                // Statistics returns whole path to file, so we need to create corresponding map key for AddFileEntry.
+                .map(addFileEntry -> Map.entry(Location.of(tableHandle.getLocation()).appendPath(addFileEntry.getPath()).toString(), addFileEntry))
```

Comment:
shall we verify `addFileEntry.getPath()` is relative to clearly indicate we don't support absolute paths yet?
(https://github.com/trinodb/trino/pull/17038)